src/librustc_typeck/collect/type_of.rs

```diff
@@ -18,6 +18,123 @@ use rustc_trait_selection::traits;
 use super::ItemCtxt;
 use super::{bad_placeholder_type, is_suggestable_infer_ty};
 
+pub(super) fn const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<DefId> {
```

Comment:
Could you add a comment explaining the general purpose of this function, with a small example?

Reply:
Added a comment referring back to the query `tcx.const_param_of`.